### PR TITLE
expression: support sequence function unfold

### PR DESCRIPTION
**Pull request by AilinKid**
#### case 1
```
create table t(a int);
create sequence seq;
insert into t values(-1),(-1),(-1);
select nextval(seq), a from t;
```
before
```
+--------------+------+
| nextval(seq) | a    |
+--------------+------+
|            1 |   -1 |
|            1 |   -1 |
|            1 |   -1 |
+--------------+------+
2 rows in set (0.01 sec)
```
after
```
+--------------+------+
| nextval(seq) | a    |
+--------------+------+
|            1 |   -1 |
|            2 |   -1 |
|            3 |   -1 |
+--------------+------+
2 rows in set (0.01 sec)
```

#### case 2
```
select lastval(seq), nextval(seq), a from t
```
before
```
+--------------+--------------+------+
| lastval(seq) | nextval(seq) | a    |
+--------------+--------------+------+
|            2 |            3 |   -1 |
|            2 |            3 |   -1 |
+--------------+--------------+------+
2 rows in set (0.00 sec)
```
after
```
+--------------+--------------+------+
| lastval(seq) | nextval(seq) | a    |
+--------------+--------------+------+
|            2 |            3 |   -1 |
|            3 |            4 |   -1 |
+--------------+--------------+------+
2 rows in set (0.00 sec)
```

### What is changed and how it works?
Add `nextval`、`lastval`、`setval` to unfold map;
Add expr list check, when the following case satisfied, it shouldn't be evaluate as vector.
- both `nextval` && `lastval` exist in expr list.
- both `nextval` && `setval` exist in expr list.
- `all` of this three exist in expr list.

Add commit-binlog sync to fix this [issue](https://github.com/pingcap/tidb/issues/15091).

### Check List 

Tests 

 - Unit test
 - Integration test

Related changes

 - Need to update the documentation

Release note

 - support sequence function unfold.

### CODE REVIEW

File: expression/chunk_executor.go [hunk 1]
```diff
@@ -30,6 +30,34 @@ func Vectorizable(exprs []Expression) bool {
 			return false
 		}
 	}
+	return HasMoreThanOneSequenceFunction(exprs)
+}
+
+// HasMoreThanOneSequenceFunction indicates where the exprs can be evaluate as a vector.
```
Comment (djshow832) [inline expression/chunk_executor.go]:
```suggestion
// HasMoreThanOneSequenceFunction indicates whether the exprs can be evaluate as a vector.
```

File: expression/chunk_executor.go [hunk 2]
```diff
@@ -30,6 +30,34 @@ func Vectorizable(exprs []Expression) bool {
...
+// When two or more of this three(nextval, lastval, setval) exists in exprs list and one of them is nextval, it should be eval row by row.
+func HasMoreThanOneSequenceFunction(exprs []Expression) bool {
```
Comment (djshow832) [inline expression/chunk_executor.go]: It seems this name doesn't conform to what it really checks.
It needs at least 2 functions.

File: ddl/sequence_test.go [hunk 3]
```diff
@@ -679,7 +679,35 @@ func (s *testSequenceSuite) TestInsertSequence(c *C) {
...
+	s.tk.MustQuery("select lastval(seq), nextval(seq), a from t1").Check(testkit.Rows("6 7 -1", "7 8 -1", "8 9 -1"))
+	s.tk.MustExec("insert into t3 select lastval(seq), nextval(seq), a from t1")
+	s.tk.MustQuery("select* from t3").Check(testkit.Rows("9 10 -1", "10 11 -1", "11 12 -1"))
+	s.tk.MustExec("delete from t3")
+
+	// if setval is folded, the result should be "101 100 -1"...
+	s.tk.MustQuery("select nextval(seq), setval(seq,100), a from t1").Check(testkit.Rows("13 100 -1", "101 <nil> -1", "102 <nil> -1"))
+	s.tk.MustExec("insert into t3 select nextval(seq), setval(seq,200), a from t1")
+	s.tk.MustQuery("select* from t3").Check(testkit.Rows("103 200 -1", "201 <nil> -1", "202 <nil> -1"))
+}
```
Comment (djshow832) [inline ddl/sequence_test.go]: Add more test cases to check more complicated situations.
e.g. `select nextval(seq)+lastval(seq), a from t1`,
`(select nextval(seq), a from t1) union (select lastval(seq), a from t1)`,
`select nextval(seq), nextval(seq), a from t1`

File: expression/chunk_executor.go [hunk 4]
```diff
@@ -30,6 +30,36 @@ func Vectorizable(exprs []Expression) bool {
 			return false
 		}
 	}
+	return checkSequenceFunction(exprs)
+}
+
+// HasMoreThanOneSequenceFunction indicates whether the exprs can be evaluated as a vector.
```
Comment (djshow832) [inline expression/chunk_executor.go]: Update the comments.

Comment (AilinKid) [reply expression/chunk_executor.go]: addressed

**Pull request merged**